pandas/core/internals/managers.py

```diff
@@ -1415,8 +1418,11 @@ class SingleBlockManager(BlockManager):
     _known_consolidated = True
     __slots__ = ()
 
-    def __init__(self, block, axis, do_integrity_check=False, fastpath=False):
-
+    def __init__(self,
+                 block: Block,
+                 axis: Union[Index, List[Index]],
```

Comment:
really? when is this an actual Index?

Reply:
Yes, it can be either. For example, when printing a DataFrame, it upplies either a index or list of indexes in various stages of the operation.

Reply:
It would probably be more consistent that it'd pick one of the two options. I'd prefer ``axis: Index``, given that this is named ``axis`` (vs. ``axes`` for ``BlockManager.__init__``).

Reply:
right, I think this was for consistency with a BlockManger where you have 2 axes; ok with changing this (followup PR); not sure how much this would take to make consistent.

Reply:
I think it was for consistency with Panel and 4dPandel etc, which isn`t relevant anymore. I`ll look into changing it.